### DISCUSSION TITLE
Is there a way to obtain errors for the barycenter coordinates and the brightness values?

### DISCUSSION THREAD

**Discussion by mapfiable** [sep-developers/sep#134] [Q&A] [answered]
The method [`sep.extract`](https://sep.readthedocs.io/en/v1.0.x/api/sep.extract.html) returns no error for the barycenter coordinates `x` and `y` or the brightness values like `flux` for example. Is there a way to obtain those? Also, since `flux` is the sum of the member pixels (in unconvolved data), does that mean only pixels that are completely within the aperture are counted? Or does it also in part count pixels that are clipped by it? Is [`sep.sum_ellipse`](https://sep.readthedocs.io/en/v1.0.x/api/sep.sum_ellipse.html) the method that is invoked for this calculation, or would it return different results if applied to the `sep.extract` data?

**Comment by kbarbary** [comment, marked as answer]
> The method [sep.extract](https://sep.readthedocs.io/en/v1.0.x/api/sep.extract.html) returns no error for the barycenter coordinates x and y or the brightness values like flux for example. Is there a way to obtain those?

The `sep.extract` documentation is misleading here: The parameters `errx2` and `erry2` are actually the variance in the barycenter coordinates `x` and `y` (the "error" squared). See [the relevant Source Extractor manual section](https://sextractor.readthedocs.io/en/latest/Position.html#position-uncertainties-errx2-erry2-errxy-erra-errb-errtheta-errcxx-errcyy-errcxy) for details.

> since flux is the sum of the member pixels (in unconvolved data), does that mean only pixels that are completely within the aperture are counted?

In the results of `sep.extract`, there are no apertures. A given pixel is either fully a member of an object or it is not. An object's member pixels can have any shape (often with jagged edges). For aperture photometry, you must use the `sep.sum_...` functions.

> Is `sep.sum_ellipse` the method that is invoked for this calculation, or would it return different results if applied to the sep.extract data?

`sep.extract` does not call any of the `sep.sum_...` functions itself.

**Reply by mapfiable** [reply]
I see, thanks a lot!
